modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -439,6 +448,34 @@ INSFVRhieChowInterpolator::finalize()
 #endif
 }
 
+void
+INSFVRhieChowInterpolator::ghostADataOnBoundary(const BoundaryID boundary_id)
+{
+  if (_a_data_provided || this->n_processors() == 1 ||
+      _velocity_interp_method == Moose::FV::InterpMethod::Average)
+    return;
+
+  // Ghost a for the elements on the boundary
+  for (auto elem_id : _moose_mesh.getBoundaryActiveLocalElemIds(boundary_id))
+  {
+    const auto & elem = _moose_mesh.elemPtr(elem_id);
+    if (elem->processor_id() != this->processor_id())
+      for (const auto i : make_range(_dim))
+        // Adding to the a coefficient will make sure the final result gets communicated
+        addToA(elem, i, 0);
```

Comment:
You should just insert into `_elements_to_push_pull` directly; then you don't loop over every dim and it's more clear what we're doing here. Alternatively you could create a new protected/private method that both this method and `addToA` can call that does thread safe insertion into `_elements_to_push_pull`.

Reply:
you need to resize _a as well.
I ll check but I think I recall this is needed

Reply:
Yea thinking about this more, I know there are find assertions in `finalize` that would break if you didn't use `addToA` here. So let's keep the call to `addToA` but you only need to do it for component 0 to make sure we create the map entry